### DISCUSSION TITLE
Cycle scratchpad window in recent order (alt+tab function of firefox).

### DISCUSSION THREAD

**Discussion by logicalbhanu** [i3/i3#4809] [Q&A]
I use scratchpad windows to keep apps which i need on almost each workspace, but many time if i have some app which i just use on scratchpad and hide it to work on main workspace app then i have to cycle through each scratchpad window to reach that  app.

Is there any way to make scratchpad cycle in alt+tab fashion i.e. in recently used order.

Some logic regarding i3ipc would also be helpful, i had tried i3ipc but issue is once i pulled the scratchpad window to any workspace to make that window visible then that window is not available in scratchpad window list, which make tracking scratchpad windows impossible.

**Comment by Airblader** [comment]
> once i pulled the scratchpad window to any workspace to make that window visible then that window is not available in scratchpad window list

This is just a rough thought, but you could use window marks to solve that in a number of ways, like marking all scratchpad windows with a shared prefix, or just the one window as it is temporarily shown, ...

**Comment by logicalbhanu** [comment]
This could work, i will try to implement it, thanks.
